server/aesdsocket.c

```diff
@@ -30,303 +30,567 @@
+               return FAILURE;
+         }
+         	
+        if(daemon)
+        {
+                pid = fork();
+                if (pid == FAILURE)
+                        return FAILURE;
+                else if (pid != 0)
+                        exit (EXIT_SUCCESS);
+                /* create new session and process group */
+                if (setsid () == FAILURE)
+                        return FAILURE;
+                /* set the working directory to the root directory */
+                if (chdir ("/") == -1)
+                        return -1;
+
+                /* redirect fd's 0,1,2 to /dev/null */
+                open ("/dev/null", O_RDWR); /* stdin */
+                dup (0); /* stdout */
+                dup (0); /* stderror */
+        }
+
+	if(daemon == false||pid == 0)
+    {
+		memset(&sev,0,sizeof(struct sigevent));
+	    /**
+        * Setup a call to timer_thread passing in the td structure as the sigev_value
+        * argument
+        */
+		td.timerParams = fd[FD_DATA];
+        sev.sigev_notify = SIGEV_THREAD;
+        sev.sigev_value.sival_ptr = &td;
+        sev.sigev_notify_function = timer_thread;
+
+        if ( timer_create(clockID,&sev,&timerID) != 0 ) 
+        {
+            printf("Error %d (%s) creating timer!\n",errno,strerror(errno));
+        }
+
+		if(!(setup_timer(clockID, timerID, 10, &startTime)))
+        {
+			printf("Timer setup error!!");
 		}
+	}
 
-		syslog(LOG_DEBUG, "Closed connection from %s\n", IP);
+    socklen_t addr_size = sizeof connection_addr;
+
+	while(!sigFlag)
+    {
 
+                //accept for a connection
+                fd[FD_CLIENT] = accept(fd[FD_SOCKET], (struct sockaddr*)&connection_addr, &addr_size);
+                
+                //if signal flag is set, leave loop
+                if(sigFlag)
+                	break;
+                	
+                //check for accept error
+                if(fd[FD_CLIENT] == -1)
+                {
+                    syslog(LOG_ERR, "ERROR: Failed to accept connection... errno:%s", strerror(errno));
+                    return FAILURE;
+                }
+                
+				//setup the values in linked list for each entry
+				linkedListPtr = malloc(sizeof(slist_data_t));
+
+                //set linked list parameters
+				(linkedListPtr->value).threadFd = fd[FD_CLIENT];
+				(linkedListPtr->value).tid = index;
+				(linkedListPtr->value).threadFlag = false;
+
+				//instert head into linked list
+        		SLIST_INSERT_HEAD(&head, linkedListPtr, entries);
+
+                //increment counter
+				index++;
+
+			//create thread and check to see if thread create failed
+			int returnVal = pthread_create(&((linkedListPtr->value).thread), NULL, threadHandler,(void*)&(linkedListPtr->value));
+			if(returnVal != 0)
+            {
+				perror("ERROR, Failed to create thread..");
+				return FAILURE;
+			}
+				
+			//check through linked list
+			SLIST_FOREACH(linkedListPtr, &head, entries) 
+            {
```

Comment:
@cchoi22915 check your tabs/spaces settings in your editor, looks like your indents are mesesed up here.  This makes your code difficult to review.